packages/atclient/src/atclient.c

```diff
@@ -432,6 +432,7 @@ int atclient_send_heartbeat(atclient *heartbeat_conn) {
   /*
    * 3. Parse response
    */
+  // how about just doing ptr == "data:ok" ?
```

Comment:
`ptr` is an address, not a value.

The value of `ptr` is a byte address that could look something like this: `0xb4aca`, so it would always return false.

You would have to use `strcmp` which compares every byte up to the null-terminator.

Reply:
The change has been made to replace existing if-statement with strcmp in [2795073](https://github.com/atsign-foundation/at_c/pull/385/commits/279507393dc0f61b49b9e177c30cef390bc3145b)